src/event.rs

```diff
@@ -35,24 +35,16 @@ use sodiumoxide::crypto::sign;
 #[derive(Clone, Eq, PartialEq)]
 pub enum Event {
     Request {
-        request        : Request,
+        request        : ExternalRequest,
         our_authority  : Authority,
         from_authority : Authority,
-        response_token : Vec<u8>,
+        response_token : SignedToken,
```

Comment:
Or you might be right, as you're passing up the same request, as a token, to be given to the generated response.